### PR TITLE
Support for Rkyv serialization in Wasmer

### PR DESCRIPTION
**Pull request by camilbancioiu**
This PR adds support for `rkyv` serialization of Wasmer instances.

The changes introduced here should be safe, because with each VM version switch, the AOTC cache is cleared by the node. Moreover, the serialized instances have no effect on consensus, so these changes cannot cause backwards incompatibility.

### CODE REVIEW

File: arwen/contexts/instanceBuilder.go [hunk 1]
```diff
@@ -4,12 +4,12 @@ import (
 	"github.com/ElrondNetwork/arwen-wasm-vm/v1_4/wasmer"
 )
 
-type wasmerInstanceBuilder struct {
+type WasmerInstanceBuilder struct {
```
Comment (sasurobert) [inline arwen/contexts/instanceBuilder.go:7]: add comment for public struct

File: wasmer/instance.go [hunk 2]
```diff
@@ -42,6 +42,14 @@ type ExportedFunctionSignature struct {
 	OutputArity int
 }
 
+func SetRkyvSerializationEnabled(enabled bool) {
```
Comment (sasurobert) [inline wasmer/instance.go:45]: add comment for public function

File: arwen/contexts/runtime.go [hunk 3]
```diff
@@ -317,6 +320,10 @@ func (context *runtimeContext) saveCompiledCode() {
 }
 
 func (context *runtimeContext) saveWarmInstance() {
+	if !warmInstancesEnabled {
```
Comment (sasurobert) [inline arwen/contexts/runtime.go:323]: delete this

File: arwen/contexts/runtime.go [hunk 4]
```diff
@@ -249,6 +248,10 @@ func (context *runtimeContext) makeInstanceFromContractByteCode(contract []byte,
 }
 
 func (context *runtimeContext) useWarmInstanceIfExists(gasLimit uint64, newCode bool) bool {
+	if !warmInstancesEnabled {
```
Comment (sasurobert) [inline arwen/contexts/runtime.go:251]: delete this

File: arwen/contexts/runtime.go [hunk 5]
```diff
@@ -24,6 +24,7 @@ var logRuntime = logger.GetOrCreate("arwen/runtime")
 var _ arwen.RuntimeContext = (*runtimeContext)(nil)
 
 const warmCacheSize = 100
+const warmInstancesEnabled = true
```
Comment (sasurobert) [inline arwen/contexts/runtime.go:27]: delete this

File: arwen/contexts/instanceBuilder.go (hunk 1 above)
Comment (camilbancioiu) [reply arwen/contexts/instanceBuilder.go:9]: Added.

File: wasmer/instance.go (hunk 2 above)
Comment (camilbancioiu) [reply wasmer/instance.go:47]: Added.

File: arwen/contexts/runtime.go (hunk 4 above)
Comment (camilbancioiu) [reply arwen/contexts/runtime.go]: Removed.

File: arwen/contexts/runtime.go (hunk 5 above)
Comment (camilbancioiu) [reply arwen/contexts/runtime.go]: Removed.

**Review by sasurobert — APPROVED** [review]

**Review by matei-p — APPROVED** [review]

**Pull request merged**